pkg/operator/reassigncontroller/reassign_controller.go

```diff
@@ -0,0 +1,103 @@
+package reassigncontroller
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/klog"
+
+	"github.com/eparis/bugzilla"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/mfojtik/bugzilla-operator/pkg/cache"
+	"github.com/mfojtik/bugzilla-operator/pkg/operator/config"
+	"github.com/mfojtik/bugzilla-operator/pkg/operator/controller"
+)
+
+type ReassignController struct {
+	context controller.ControllerContext
+	config  *config.OperatorConfig
+}
+
+func NewReassignController(ctx controller.ControllerContext, operatorConfig config.OperatorConfig, recorder events.Recorder) factory.Controller {
+	c := &ReassignController{context: ctx, config: &operatorConfig}
+	return factory.New().WithSync(c.sync).ResyncEvery(1*time.Hour).ToController("ReassignController", recorder)
+}
+
+func (c *ReassignController) sync(ctx context.Context, syncCtx factory.SyncContext) (err error) {
+	client := c.context.NewBugzillaClient(ctx)
+	assigneeNotifiedBugs, err := getAssigneeNotifiedBugsList(client, c.config)
+	if err != nil {
+		syncCtx.Recorder().Warningf("BuglistFailed", err.Error())
+		return err
+	}
+
+	for _, bug := range assigneeNotifiedBugs {
+		history, err := client.GetCachedBugHistory(bug.ID, bug.LastChangeTime)
+		if err != nil {
+			syncCtx.Recorder().Warningf("GetBugFailed", err.Error())
+			continue
+		}
+		var (
+			whenNotified         time.Time
+			whenComponentChanged *time.Time
+		)
+		for _, h := range history {
+			changedAt, err := time.Parse(time.RFC3339, h.When)
+			if err != nil {
+				klog.Errorf("unable to parse change time %q for bug %+v: %v", h.When, bug, err)
+				continue
+			}
+			for _, change := range h.Changes {
+				switch change.FieldName {
+				case "cf_devel_whiteboard":
+					whenNotified = changedAt
```

Comment:
this needs more. You have to check that AssignedModified is in the new value, but not in the old.

Are we sure history is ordered?

Reply:
@sttts we can try to sort the slice based on when beforehand to be sure.

Reply:
Or just compare whether whenNotified < changedAt. You want the last one.